docker-java-core/src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java

```diff
@@ -402,6 +410,15 @@ public BuildImageCmd withTarget(String target) {
         return this;
     }
 
+    @Override
+    public BuildImageCmd withExtraHost(String extraHost) {
+        if (this.extraHosts == null) {
+            this.extraHosts = new HashSet<>();
+        }
+        this.extraHosts.add(extraHost);
```

Comment:
set instead add?

Reply:
The method parameter `extraHost` is of type String we will want to _add_ to a `Set`. Can you elaborate? I don't think I understand your comment.

Reply:
It would be good to keep transparent with as setter, better add additional method `addExtraHost(String)` if it really needed

Reply:
I orientated myself on this method for `buildArgs`: https://github.com/docker-java/docker-java/blob/master/docker-java-api/src/main/java/com/github/dockerjava/api/command/BuildImageCmd.java#L193. Granted, it's a `Map` but that doesn't really matter too much. Would you be opposed with keeping the method I introduced and then add another one for providing a `Set`?

Reply:
Never mind, misread your comment. Will make that change.

Reply:
Made the change. Decided to only expose a method for a `Set<String>`.